pkg/workload/workload.go

```diff
@@ -348,14 +348,19 @@ func (i *Info) IsUsingTAS() bool {
 		})
 }
 
-// IsRequestingTAS returns information if the workload is requesting TAS
-func (i *Info) IsRequestingTAS() bool {
+// IsExplicitlyRequestingTAS returns information if the workload is requesting TAS
+func (i *Info) IsExplicitlyRequestingTAS() bool {
 	return slices.ContainsFunc(i.Obj.Spec.PodSets,
 		func(ps kueue.PodSet) bool {
-			return ps.TopologyRequest != nil
+			return IsTASExplicitlyRequested(&ps)
 		})
 }
```

Comment:
```suggestion
func IsExplicitlyRequestingTAS(ps *kueue.PodSet) bool {
	return slices.ContainsFunc(i.Obj.Spec.PodSets,
		func(ps kueue.PodSet) bool {
			tr := ps.TopologyRequest
			return tr != nil && (tr.Unconstrained != nil || tr.Required != nil || tr.Preferred != nil || tr.PodSetSliceRequiredTopology != nil || tr.PodSetSliceSize != nil)
		})
}
```

Exposing such almost same semantic helpers (`IsTASExplicityRequested` and `(i *Info) IsExplicitlyRequestingTAS)` is not good way. That will bring us confusion and lack of maintainability.

So, we should have only IsExplicitlyRequestingTAS w/o the receiver since it actually does not use Workload Information.

Reply:
> I agree it requires improvement, especially since the naming is practically the same. The difference between those two is that one iterates over the Workload's PodSets so it actually uses its information, and the second is only about a particular PodSet. Let me think how to refactor it in the cleanest way

I see. What about the following? Couldn't we combine those two functions into one?

```go
func IsExplicitlyRequestingTAS(podSets ...kueue.PodSet) bool {
	return slices.ContainsFunc(podSets,
		func(ps kueue.PodSet) bool {
			tr := ps.TopologyRequest
			return tr != nil && (tr.Unconstrained != nil || tr.Required != nil || tr.Preferred != nil || tr.PodSetSliceRequiredTopology != nil || tr.PodSetSliceSize != nil)
		})
}
```

Reply:
The above my proposal could be called by `workload.IsExplicitlyRequestingTAS(ps)` even when ps typed is `kueue.PodSet`. Doesn't that work well like this one:

```go
// isTASImplied returns true if TAS is requested implicitly
func isTASImplied(ps *kueue.PodSet, cq *cache.ClusterQueueSnapshot) bool {
	return !workload.IsExplicitlyRequestingTAS(*ps) && cq.IsTASOnly()
}

// isTASRequested checks if TAS is requested for the input PodSet, either
// explicitly or implicitly.
func isTASRequested(ps *kueue.PodSet, cq *cache.ClusterQueueSnapshot) bool {
	return workload.IsExplicitlyRequestingTAS(*ps) || isTASImplied(ps, cq)
}
```